psychopy/experiment/components/_base.py

```diff
@@ -338,50 +338,58 @@ def writeStartTestCode(self, buff, extra=""):
             `%(key)s` syntax to insert the values of any necessary params. Default is an empty 
             string.
         """
+        # create copy of params dict so we can change stuff without harm
+        params = self.params.copy()
 
         # Get starting indent level
         startIndent = buff.indentLevel
 
-        if self.params['startVal'].val in ('', None, -1, 'None'):
-            # If we have no start time, don't write start code
-            return buff.indentLevel - startIndent
+        if params['startVal'].val in ('', None, -1, 'None'):
+            if extra:
+                # if we have extra and no stop condition, extra is the only stop condition
+                params['startType'] = params['startType'].copy()
```

Comment:
Is this extra `copy()` necessary? Isn't `params` already a copy that we can change without harm, so should we simply do:
```
params['startType'] = "condition"
params['startVal'] = "False"
```
here?

Reply:
`params` is a shallow copy of `self.params`, so the `Param` objects contained within `params` are the same objects in `self.params`. What you suggest would work in terms of not affecting `self.params`, but there's other points in here where `params['startVal'].val` are referenced, so if we did `params['startVal'] = "False"` then these would fail as the Param object is replaced by a string.

Reply:
Ah you are right! I forgot PsychoPy uses a dedicated `Params` class object to store parameters rather than simple key-value pairs in a dictionary. Got it - this is what you meant by deep copying. Thanks a lot for explaining this! Looks good to me!